src/renderers/webgl/WebGLRenderLists.js

```diff
@@ -137,13 +137,32 @@ function WebGLRenderList() {
 
 	}
 
+	// remove any dangling references to make sure no memory is leaked
+	// when the count of objects in a scene is reduced
+	function cleanup() {
+
+		for ( var i = renderItemsIndex; i < renderItems.length; i ++ ) {
+
+			var renderItem = renderItems[ i ];
+			if ( renderItem.id === null ) break;
```

Comment:
Can you explain this line?

Reply:
It's not necessary to iterate over the entire `renderItems` array. As soon as you detect a render item with `id` equals null, you can stop processing since the current and all subsequent render items are already invalidated.

Reply:
And when does `renderItem.id` become `null`?

Reply:
When it was reset by this function in previous frame. The renderItem slot is "cleaned", but still ready to use. Once any item contains null, we can be sure all followind items contain nulls as well.

Reply:
At least this is how it should work. Another fix was needed for that (I am setting the id = null now).

Reply:
Overall it seems #12464 was in a better shape. I am ashamed on how many fixes were needed in such a short and seemingly trivial code.

Reply:
The problem was that you though this was a trivial code 😁
This is a very delicate piece of code...